streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamPartitionAssignor.java

```diff
@@ -424,9 +460,41 @@ public Subscription subscription(Set<String> topics) {
         // change log topics should be compacted
         prepareTopic(stateChangelogTopicToTaskIds, true /* compactTopic */, true);
 
+        Map<String, Assignment> assignment = new HashMap<>();
+        for (AssignmentDetail assignmentDetail : assignmentDetails) {
+            assignment.put(assignmentDetail.consumer, assignmentDetail.createAssignment());
+        }
         return assignment;
     }
 
+    class AssignmentDetail {
```

Comment:
nit: rename to `AssignmentInfo` as with `SubscriptionInfo` since it is also per-consumer?